### DISCUSSION TITLE
`vercel env` needs improvements for continuous integration

### DISCUSSION THREAD

**Discussion by 0xJem** [vercel/vercel#7565] [Ideas]
Aim:
- Secrets managed in GitHub
- Pushed to Vercel using the CLI through GitHub Actions

What prevents this:
- `vercel env add` has no option to replace an existing variable, so the existing variable needs to be removed first.
- `vercel env rm` returns a non-zero exit code if the named variable does not exist. There should be a `--force` flag to avoid this.
- There is no way to load variables in bulk (e.g. from a .env file) using `vercel env add`
- If you are loading a variable's value from a file, the environment (preview/production) needs to be specified as well. This doesn't make sense.

Suggestions:
- `vercel env add` should have a `--replace` flag
- `vercel env add` should allow for loading variables from a file, e.g. `vercel env add [environment] < [file]`
- Loading a variable's value from a file should not require the environment, e.g. `vercel env add [name] < [file]`

**Comment by spencerlepine** [comment]
Temporary fix: `vercel env rm MONGO_URI production -y || echo "Failed to remove env var"`

1. Remove env var if it exists
2. Set env var with latest value
```yml
- name: "Deploy application"
  run: |
    npm i vercel -g
    # echo [value] | vercel env add [name] [environment]
    vercel --token ${{ secrets.VERCEL_TOKEN }} env rm MONGO_URI production -y || echo "Failed to remove env var"
    mongoURI="${{ secrets.MONGO_URI_IMAGES }}"
    echo $mongoURI | vercel --token ${{ secrets.VERCEL_TOKEN }} env add MONGO_URI production
    vercel --token ${{ secrets.VERCEL_TOKEN }} --prod
```
